portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,16 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/**
+ * Fetches the data from our Java servlet 
+ */
+const getData = async () => {
+    const res = await fetch('/data');
+    const data = await res.text();
+
+    document.getElementById('fetch-container').innerText = data;
+}
+
+// Gets called automatically
+getData();
```

Comment:
You can also use onload="getData()" to your fetch-container div instead of calling it here (see "you can use the onload attribute of the <body> element to fetch the content when the page loads" comment in the walkthrough). Docs: https://www.w3schools.com/jsref/event_onload.asp

Reply:
You are totally right, completely forgot about this.